.github/workflows/buildAndTest.yml

```diff
@@ -0,0 +1,71 @@
+# Copyright 2022 The StableHLO Authors.
+
+name: Build and Test
+
+on:
+  pull_request:
+    branches: [ main ]
+  push:
+    branches: [ main ]
+  schedule:
+    # Always regenerate once every day
+    - cron:  '* 12 * * *'
+  workflow_dispatch:
+
+# Ensure that only a single job or workflow using the same
+# concurrency group will run at a time. This would cancel
+# any in-progress jobs in the same github workflow and github
+# ref (e.g. refs/heads/main or refs/pull/<pr_number>/merge).
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+# Build using CMake and run tests.
+# Use Cached LLVM to improve build times.
+# Current setup only builds on linux.
+jobs:
+  build-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os-arch: [maci-x86_64, ubuntu-x86_64]
+        include:
+          - os-arch: ubuntu-x86_64
+            os: ubuntu-22.04
+            llvm-build-dir: "$GITHUB_WORKSPACE/llvm-build"
+            stablehlo-build-dir: "$GITHUB_WORKSPACE/build"
+          - os-arch: maci-x86_64 
+            os: macos-latest
+            llvm-build-dir: "$GITHUB_WORKSPACE/llvm-build-maci"
+            stablehlo-build-dir: "$GITHUB_WORKSPACE/build-maci"
```

Comment:
Why are Linux and macOS directories different?

Reply:
Removed. I think I was dodging the risk that there may be some sort of NFS thing going on with shared directories between the two platforms, but it doesn't seem like that's the case with GH actions.